### ISSUE TITLE
Feature Suggestion - Mark for Above the Fold

### ISSUE THREAD

**Issue by pingram3541** [pojome/elementor#2203]
I know we can use plugins like https://wordpress.org/plugins/above-the-fold-optimization/

I was just thinking it would be cool if we could mark a section in the advanced area as "above the fold" and the css style output when processed shows on the front end as an inline style tag in the `head` section.

**Comment by optimizewp** [comment]
@pingram3541  You are right. The code is already there, and the inlined one will be printed with the @media queries too. But, this would work as Critical CSS only if all the theme was made with Elementor and rendered in the same edited page. This is mostly not the case. Main part of the page layout are built by the theme, and we know theme developers are not interested in adding these kind of features. The most part of them promote their themes as the fastest one, and their demo are really bad performing. Pojo themes demos are a rare case of well optimized demo sites ;)
It could be useful, though! Nice request!

**Comment by pingram3541** [comment]
@optimizewp Thanks Paul, I had thought about that but with the many tools out there that allow for Elementor to provide custom header/footer areas as well as the Canvas template, this would be very useful and save a lot of time not having to dissect a production ready theme + plugin css/js output, additionally and then have to reverse that every time we want to make edits to the original default (non-optimized) output, rinse repeat, very exhaustive and not efficient.

Of course, those of us that know how to patch a poorly written theme to include all header styles inline, even dynamic css.php type theme options, can get around this but that still leaves the page builder plugins to also catch up to par with modern needs.

**Comment by shimsa** [comment]
What's the status of this feature request now?
I notice, that Featured Image loaded like a background with Dynamic Tags feature, is printed like inline CSS (elementor-frontend-inline-css). But even better would be:
- section with HTML Tag "header" -> print inline CSS automatically, because is always top of the site
- add switch button "Critical CSS" to all sections -> web designer choose which CSS code want to inline
- inline CSS "elementor-frontend-inline-css" move before all <link rel='stylesheet'...

**Comment by ayoubkhan558** [comment]
+1
